sophia/src/term/literal/_convert.rs

```diff
@@ -0,0 +1,495 @@
+//! Public exported by parent-module `literal`.
+//!
+
+use super::*;
+use crate::ns::xsd;
+use crate::term::{Iri, Result, StaticTerm, TermData, TermError};
+use crate::triple::stream::{SinkError, SourceError, StreamResult};
+
+/// Provide an IRI to specify the semantic datatype.
+pub trait DataType {
+    /// Static datatype of this Rust-type.
+    fn dt() -> &'static Iri<&'static str>;
+
+    /// Datatype of `self`.
+    ///
+    /// This can be used for `enum`s where each variant may has its own
+    /// datatype.
+    fn dt_of(&self) -> &'static Iri<&'static str> {
```

Comment:
Might look strange when you call it with the `.` notation: `x.dt_of()`.
I suggest renaming it `dt`, or better `datatype`.